swarm/api/api.go

```diff
@@ -350,77 +350,54 @@ func (a *API) Get(ctx context.Context, manifestAddr storage.Address, path string
+				return reader, mimeType, status, nil, err
+			}
+
+			// validate that data as multihash
+			decodedMultihash, err := multihash.FromMultihash(rsrcData)
+			if err != nil {
+				apiGetInvalid.Inc(1)
+				status = http.StatusUnprocessableEntity
+				log.Warn("invalid resource multihash", "err", err)
+				return reader, mimeType, status, nil, err
+			}
+			manifestAddr = storage.Address(decodedMultihash)
+			log.Trace("resource is multihash", "key", manifestAddr)
 
-			// use this key to retrieve the latest update
-			params := mru.LookupLatest(rootAddr)
-			rsrc, err = a.resource.Lookup(ctx, params)
+			// get the manifest the multihash digest points to
+			trie, err := loadManifest(ctx, a.fileStore, manifestAddr, nil)
 			if err != nil {
 				apiGetNotFound.Inc(1)
 				status = http.StatusNotFound
-				log.Debug(fmt.Sprintf("get resource content error: %v", err))
+				log.Warn(fmt.Sprintf("loadManifestTrie (resource multihash) error: %v", err))
 				return reader, mimeType, status, nil, err
 			}
 
-			// if it's multihash, we will transparently serve the content this multihash points to
-			// \TODO this resolve is rather expensive all in all, review to see if it can be achieved cheaper
-			if rsrc.Multihash() {
-
-				// get the data of the update
-				_, rsrcData, err := a.resource.GetContent(rootAddr)
-				if err != nil {
-					apiGetNotFound.Inc(1)
-					status = http.StatusNotFound
-					log.Warn(fmt.Sprintf("get resource content error: %v", err))
-					return reader, mimeType, status, nil, err
-				}
-
-				// validate that data as multihash
-				decodedMultihash, err := multihash.FromMultihash(rsrcData)
-				if err != nil {
-					apiGetInvalid.Inc(1)
-					status = http.StatusUnprocessableEntity
-					log.Warn("invalid resource multihash", "err", err)
-					return reader, mimeType, status, nil, err
-				}
-				manifestAddr = storage.Address(decodedMultihash)
-				log.Trace("resource is multihash", "key", manifestAddr)
-
-				// get the manifest the multihash digest points to
-				trie, err := loadManifest(ctx, a.fileStore, manifestAddr, nil)
-				if err != nil {
-					apiGetNotFound.Inc(1)
-					status = http.StatusNotFound
-					log.Warn(fmt.Sprintf("loadManifestTrie (resource multihash) error: %v", err))
-					return reader, mimeType, status, nil, err
-				}
-
-				// finally, get the manifest entry
-				// it will always be the entry on path ""
-				entry, _ = trie.getEntry(path)
-				if entry == nil {
-					status = http.StatusNotFound
-					apiGetNotFound.Inc(1)
-					err = fmt.Errorf("manifest (resource multihash) entry for '%s' not found", path)
-					log.Trace("manifest (resource multihash) entry not found", "key", manifestAddr, "path", path)
-					return reader, mimeType, status, nil, err
-				}
-
-			} else {
-				// data is returned verbatim since it's not a multihash
-				return rsrc, "application/octet-stream", http.StatusOK, nil, nil
+			// finally, get the manifest entry
+			// it will always be the entry on path ""
```

Comment:
This is the resource manifest, not the bzz manifest. Resolving sub-paths is the responsibility of the bzz manifest, so no need to resolve paths here.
The reason it is *always the entry on path ""* is because you're reusing the common manifest structure for resource manifests and you only need one entry

Reply:
So with this current code I can still specify a subpath on the manifest the resource points to when I retreive using a MRU manifest?

Reply:
With the current code (and what is currently in `master`) you can:

`GET /bzz:/<resource_manifest_hash>/path/to/my/file.txt`
and use
`GET /bzz-resource:/<resource_manifest_hash>`
... to get the raw data of the resource (probably a `0x1b20xxx` multihash). This call will return a (small) octet stream
Also, 
`GET /bzz-resource:/<resource_manifest_hash>/<period>`
`GET /bzz-resource:/<resource_manifest_hash>/period/version`
and
`GET /bzz-resource:/<resource_manifest_hash>/meta` to read info about what the node knows about the resource.

Resolving paths under `/bzz-resource:/` was never supported (and also does not make sense in my opinion. Filesystem abstraction is `bzz:`'s domain

Reply:
Ok I wasn't actually aware that we could do this for the `bzz` scheme. Great!

Reply:
note that the following:
`GET /bzz-resource:/<resource_manifest_hash>/<period>`
`GET /bzz-resource:/<resource_manifest_hash>/period/version`
has changed in #839 in favor of query strings after a second major cleanup